pkg/streamingpromql/engine_test.go

```diff
@@ -1444,6 +1444,21 @@ func TestAnnotations(t *testing.T) {
 			},
 		},
 
+		"binary operation between native histograms with exponential and custom buckets": {
+			data: nativeHistogramsWithCustomBucketsData,
+			expr: `metric{series="exponential-buckets"} + ignoring(series) metric{series="custom-buckets-1"}`,
+			expectedWarningAnnotations: []string{
+				`PromQL warning: vector contains a mix of histograms with exponential and custom buckets schemas for metric name "" (1:1)`,
```

Comment:
Note to reviewers: these messages look weird given we have no metric name in the case of binary operations, but it's what Prometheus does as well, so I've kept it the same for the sake of consistency.

Reply:
We should open an upstream issue to fix it there and then eventually in MQE

Reply:
Yep, done: https://github.com/prometheus/prometheus/issues/14669